src/Symfony/Component/Security/Core/Authorization/Voter/Voter.php

```diff
@@ -30,7 +30,11 @@ public function vote(TokenInterface $token, $subject, array $attributes)
         $vote = self::ACCESS_ABSTAIN;
 
         foreach ($attributes as $attribute) {
-            if (!$this->supports($attribute, $subject)) {
+            try {
+                if (!$this->supports($attribute, $subject)) {
+                    continue;
+                }
+            } catch (\TypeError $e) {
```

Comment:
You cannot be sure that the `supports()` call is the source of the `TypeError`. The voter implementation might also have triggered it and in that case we must not destroy it here. You can find some examples how to handle those cases inside the PropertyAccess component.

Reply:
For reference:

https://github.com/symfony/symfony/blob/6d4b26d25f3dcc224d7c66eb1e3bd1665c0bea21/src/Symfony/Component/PropertyAccess/PropertyAccessor.php#L191-L216

Reply:
@derrabus Given that if the type of the `@attribute` argument is valid for the `supports()` method, in this case, it will also be valid for the method `voteOnAttribute()`, so, there is no need to broadly more embed the code with the `try/catch`. However, if you think it is still better to include all the contents of the `foreach` or the entire the `foreach`, I will modify the PR.

@wouterj Thanks for the reference, I deduce that you want to throw an exception if the regex does not correspond exactly to the typing problem concerned. Is that it?

Reply:
@francoispluchino You got me wrong, I'm afraid. It's perfectly fine if you only wrap the `supports()` call into the `try` block.

However, if you catch a `TypeError`, it could result from the `supports()` call with a non-string attribute (this is the case you're interested in), but it could also be triggered further down the call stack: The `supports()` implementation could have called a method that called a method that triggered a `TypeError` for completely unrelated reasons. In the latter case, the `TypeError` has nothing to do with the problem you're solving and thus must be rethrown.

Reply:
@francoispluchino yes, I think that's also what @derrabus is trying to explain. The implementation of `support()` may call some other functions with wrong types - resulting in a `TypeError` that was not caused by `support(string $attribute)`. In these cases, we still want to have the error and not silently ignore it. So we should check the error message of `TypeError` to see if it's the `support(string $attribute)` call that caused it.

Reply:
Indeed, I am working on the regex to take into account only the case which concerns us and to throw the exception if it is not the case.

Reply:
Modification carried out and test added.